source/parser/function.civet

```diff
@@ -661,6 +661,7 @@ function iterationDeclaration(statement: IterationStatement | ForStatement)
     bindings: []
   }
 
+  object := statement.type is "ForStatement" and statement.object
```

Comment:
I like your idea of allowing any iteration! So I think you can remove `statement.type is "ForStatement"`. Also, this probably makes sense to move up to where `reduction` is defined.